packages/edge-config/src/utils/swr-fn.ts

```diff
@@ -8,18 +8,20 @@ import { clone } from './index';
 export const swr = trace(
   function swr<T extends (...args: any[]) => Promise<any>>(fn: T): T {
     let latestInvocationId = 0;
-    let staleValuePromise: null | Promise<unknown> = null;
+    const staleValuePromiseMap = new Map<string, Promise<unknown>>();
 
     return (async (...args: any[]) => {
+      const argsKey = JSON.stringify(args);
       const currentInvocationId = ++latestInvocationId;
+      const staleValuePromise = staleValuePromiseMap.get(argsKey);
```

Comment:
Would the `currentInvocationId` need to be per argsKey? Otherwise the cache might not get set later on

Reply:
I think we could refactor this whole thing to get rid of the `swr` function. Instead of fetching individual items, we want to fetch the whole edge config in dev mode. We could add this to `index.ts` directly, since it already deals with "get the right info out of all edge config items when a function like get is called"

Reply:
Good point. I removed the counter to just compare the promise. We should only overwrite it, if it's still the same